File: chapter_03/python/3_1.py
list_a = [1, 23, 542, 656, 88, 4, 3, 6, 4, 33, 32, 23] #indexは0~11
target_num = 23

max_index = 0
for index in range(len(list_a)):
    if list_a[index] == target_num:
        max_index = index

print(f"max_index : {max_index}") # 答えは11

"""
計算量O(N)
"""

File: chapter_03/c++/3_1.cpp
#include <iostream>

using namespace std;

int main() {
    int target_num = 23;
    int list_a[] = {1, 23, 542, 656, 88, 4, 3, 6, 4, 33, 32, 23};

    int max_idx = 0;
    for (int i = 0; i < sizeof(list_a) / sizeof(list_a[0]); i++) {
        if (list_a[i] == target_num) {
            max_idx = i;
        }
    }
    cout <<"最大のインデックスは、" << max_idx << endl;
}